Tests/Unit/NetworkServer/LoRaDeviceApiServiceTests.cs

```diff
@@ -124,6 +126,39 @@ public async Task SearchByEuiAsync_DevEui_Parses_Json(string json, string expect
             Assert.Equal(expected, result);
         }
 
+        [Fact]
+        public async Task FetchStationFirmwareAsync_Returns_Stream_And_Content_Length()
+        {
+            var contentBytes = Encoding.UTF8.GetBytes("Test");
+            using var memorystream = new MemoryStream(contentBytes.Length);
+            await memorystream.WriteAsync(contentBytes);
+            memorystream.Seek(0, SeekOrigin.Begin);
+            // arrange
+            var facadeMock = new Mock<IServiceFacadeHttpClientProvider>();
+            using var httpHandlerMock = new HttpMessageHandlerMock();
+            httpHandlerMock.SetupHandler(r =>
+            {
+                var response = new HttpResponseMessage(System.Net.HttpStatusCode.OK)
+                {
+                    Content = new StreamContent(memorystream),
+
+                };
+                return response;
+            } );
```

Comment:
You can simplify by returning a `ByteArrayContent` in the response from the mock:

```suggestion
            var contentBytes = Encoding.UTF8.GetBytes("Test");
            // arrange
            var facadeMock = new Mock<IServiceFacadeHttpClientProvider>();
            using var httpHandlerMock = new HttpMessageHandlerMock();
            httpHandlerMock.SetupHandler(_ => new HttpResponseMessage(System.Net.HttpStatusCode.OK)
            {
                Content = new ByteArrayContent(contentBytes),
            });
```